### PR TITLE
[CARBONDATA-649] fix for update with rand function

### PR DESCRIPTION
**Pull request by ashwini-krishnakumar**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[CARBONDATA-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).
 - [ ] Testing done
 
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - What manual testing you have done?
        - Any additional information to help reviewers in testing this change.
         
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
                 
---

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by ravipesala** [comment]
retest this please

File: integration/spark2/src/main/scala/org/apache/spark/sql/optimizer/CarbonLateDecodeRule.scala [hunk 1]
```diff
@@ -841,3 +868,45 @@ case class CarbonDecoderRelation(
 
   lazy val dictionaryMap = carbonRelation.carbonRelation.metaData.dictionaryMap
 }
+
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/optimizer/CarbonLateDecodeRule.scala]: Move this class dummy class out of this class

**Comment by ashwini-krishnakumar** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
SDV Build Success with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/364/

**Comment by ashwini-krishnakumar** [comment]
@ravipesala please review

**Comment by ravipesala** [comment]
LGTM

**Comment by ravipesala** [comment]
@ashwini-krishnakumar Please amend all comments as single comment. There is an issue while merging to master.

**Comment by ravipesala** [comment]
@ashwini-krishnakumar Please amend all comments as single comment. There is an issue while merging to master.

**Comment by ravipesala** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/596/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/UpdateCarbonTableTestCase.scala [hunk 2]
```diff
@@ -434,6 +434,43 @@ class UpdateCarbonTableTestCase extends QueryTest with BeforeAndAfterAll {
...
+    sql(
+      s"""CREATE TABLE if not exists rand (CUST_ID int,CUST_NAME String,ACTIVE_EMUI_VERSION string,
+         | DOB timestamp, DOJ timestamp, BIGINT_COLUMN1 bigint,BIGINT_COLUMN2 bigint,
+         | DECIMAL_COLUMN1 decimal(30,10), DECIMAL_COLUMN2 decimal(36,10),Double_COLUMN1 double,
+         | Double_COLUMN2 double,INTEGER_COLUMN1 int) STORED BY 'org.apache.carbondata.format'
+         | TBLPROPERTIES('DICTIONARY_INCLUDE'='CUST_ID,CUST_NAME,ACTIVE_EMUI_VERSION,DOB,DOJ,BIGINT_COLUMN1,
+         | BIGINT_COLUMN2,DECIMAL_COLUMN1,DECIMAL_COLUMN2,Double_COLUMN1,Double_COLUMN2,INTEGER_COLUMN1')"""
+        .stripMargin).collect()
+
+    sql(s"""LOAD DATA INPATH '$resourcesPath/IUD/2000_UniqData.csv' into table rand OPTIONS('DELIMITER'=',' , 'QUOTECHAR'='"','BAD_RECORDS_ACTION'='FORCE','FILEHEADER'='CUST_ID,CUST_NAME,ACTIVE_EMUI_VERSION,DOB,DOJ,BIGINT_COLUMN1,BIGINT_COLUMN2,DECIMAL_COLUMN1,DECIMAL_COLUMN2,Double_COLUMN1,Double_COLUMN2,INTEGER_COLUMN1')""").collect
```
Comment (ravipesala) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/UpdateCarbonTableTestCase.scala]: Try to use the csv files which are already existed for this testcase. I don't see any reason to add new csv for this simple testcase.

**Comment by ashwini-krishnakumar** [comment]
Updated UpdateCarbonTableTestCase.scala to use an existing csv file

**Comment by ravipesala** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/653/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**